spec/rubocop/cop/style/negated_if_spec.rb

```diff
@@ -1,90 +1,207 @@
+                      '  some_method',
+                      'end',
+                      'if not condition or another_condition',
+                      '  some_method',
+                      'end',
+                      'some_method if not condition or another_condition'])
+      expect(cop.offenses).to be_empty
+    end
+
+    it 'accepts an if where the condition is doubly negated' do
+      inspect_source(cop,
+                     ['if !!condition',
+                      '  some_method',
+                      'end',
+                      'some_method if !!condition'])
+      expect(cop.offenses).to be_empty
+    end
+
+    it 'is not confused by negated elsif' do
+      inspect_source(cop,
+                     ['if test.is_a?(String)',
+                      '  3',
+                      'elsif test.is_a?(Array)',
+                      '  2',
+                      'elsif !test.nil?',
+                      '  1',
+                      'end'])
+
+      expect(cop.offenses).to be_empty
+    end
+
+    it 'autocorrects by replacing if not with unless' do
+      corrected = autocorrect_source(cop, 'something if !x.even?')
+      expect(corrected).to eq 'something unless x.even?'
+    end
+
+    it 'autocorrects by replacing parenthesized if not with unless' do
+      corrected = autocorrect_source(cop, 'something if (!x.even?)')
+      expect(corrected).to eq 'something unless (x.even?)'
+    end
+
+    it 'autocorrects by replacing unless not with if' do
+      corrected = autocorrect_source(cop, 'something unless !x.even?')
+      expect(corrected).to eq 'something if x.even?'
+    end
   end
 
-  it 'accepts an if where the condition is doubly negated' do
-    inspect_source(cop,
-                   ['if !!condition',
-                    '  some_method',
-                    'end',
-                    'some_method if !!condition'])
-    expect(cop.offenses).to be_empty
+  describe 'with “prefix” style' do
+    subject(:cop) do
+      config = RuboCop::Config.new(
+        'Style/NegatedIf' => {
+          'SupportedStyles' => %w(both prefix postfix),
+          'EnforcedStyle' => 'prefix'
+        }
+      )
+
+      described_class.new(config)
+    end
+
+    it 'registers an offence for prefix' do
+      inspect_source(cop,
+                     ['if !foo',
+                      'end'])
+
+      expect(cop.messages).to eq(
+        ['Favor `unless` over `if` for negative conditions.']
+      )
+    end
+
+    it 'does not register an offence for postfix' do
+      inspect_source(cop, 'foo if !bar')
+
+      expect(cop.offenses).to be_empty
+    end
+
+    it 'autocorrects for prefix' do
+      corrected = autocorrect_source(cop,
+                                     ['if !foo',
+                                      'end'])
+
+      expect(corrected).to eq [
+        'unless foo',
+        'end'
+      ].join("\n")
+    end
+
+    it 'does not autocorrect for postfix' do
```

Comment:
Ah! So it's not necessary (or even possible) to test the not case?

@Drenmi So I guess that means the conditionals don't need to be added to the `autocorrect` method, and that explains why the tests that I expected to fail passed, because if it only autocorrects when there's an offense, then the `on_if` method acts like a guard.